src/sphinx_book_theme/__init__.py

```diff
@@ -245,11 +222,20 @@ def setup(app: Sphinx):
     app.connect("builder-inited", check_deprecation_keys)
     app.connect("builder-inited", update_sourcename)
     app.connect("builder-inited", update_context_with_repository_info)
-    app.connect("builder-inited", update_general_config)
     app.connect("html-page-context", add_metadata_to_page)
     app.connect("html-page-context", hash_html_assets)
     app.connect("html-page-context", update_templates)
 
+    # This extension has both theme-like and extension-like features.
+    # Themes are initialised immediately before use, thus we cannot
+    # rely on an event to set the config - the theme config must be
+    # set in setup(app):
+    update_general_config(app)
+    # Meanwhile, extensions are initialised _first_, and any config
+    # values set during setup() will be overwritten. We must therefore
+    # register the `config-inited` event to set these config options
+    app.connect("config-inited", update_general_config)
```

Comment:
I thought that for some folks if the config initialization was called twice, it would raise an error. If that's not the case then I think it's fine to just do this twice as long as nobody reports regressions

Reply:
@choldgraf to be clear, I don't think this should ever set the config twice. AFAICT either:
1. SBT is _only_ used as a theme, which means `config-inited` has already triggered by the time we register the listener
2. SBT is _both_ a theme and extension, in which case the `app.config` in `setup()` is nearly immediately clobbered.

I'm hoping _that_ isn't a cause of bugs, and I've not seen any evidence as yet to suggest it might be :crossed_fingers: